Explain rules on defining Kapitan dependencies (#114)
projectsyn/documentation: docs/modules/ROOT/pages/references/style-guide.adoc

* Put a  https://www.doxygen.nl/manual/docblocks.html[Docblock] at the top of each Jsonnet file or library to indicate its purpose.
* Exceptions can be made for `app.jsonnet` and `main.jsonnet`.

== Commodore component style

=== Defining dependencies

> * Download external dependencies to a path within the components directory.

Reviewer: ```suggestion
* Download external dependencies to a path within the component's directory.
```

> * Add that dirctory to the components `.gitignore` file.

Reviewer: ```suggestion
* Add that directory to the component's `.gitignore` file.
```

> * Ensure the outpt path changes when the upstream change.

Reviewer: ```suggestion
* Ensure the output path changes when the upstream changes.
```
